transports/janus_mqtt.c

```diff
@@ -306,6 +379,25 @@ int janus_mqtt_init(janus_transport_callbacks *callback, const char *config_path
 	}
 
 	/* Connect configuration */
+	janus_config_item *mqtt_version = janus_config_get(config, config_general, janus_config_type_item, "mqtt_version");
+	const char *mqtt_version_str = (mqtt_version && mqtt_version->value) ? g_strdup(mqtt_version->value) : JANUS_MQTT_VERSION_DEFAULT;
```

Comment:
No need to `g_strdup` here, you're not keeping `mqtt_version_str`: it's only declared and used here. The way it is now you're introducing a memory leak.